std/haxe/math/bigint/BigInt_.hx

```diff
@@ -0,0 +1,876 @@
+/*
+ * Copyright (C)2005-2022 Haxe Foundation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package haxe.math.bigint;
+
+import haxe.ds.Vector;
+import haxe.io.Bytes;
+
+/* Original code courtesy Chuck Batson (github.com/cbatson) */
+@:allow(unit)
+@:allow(haxe.math.bigint)
+class BigInt_ {
```

Comment:
`@:allow` has no effect on public classes.

Classes not meant for access outside the package should preferably be added to other modules as private classes (i.e. in this case to BigInt.hx) or at least have the  `@:noCompletion` and `@:noDoc` annotations added to prevent the types from being suggested by the completion server and from showing up in the API docs.

Reply:
> @:allow has no effect on public classes

It does? It's about giving access to its private fields.
But it won't allow access to a private class from outside its module if that's what you're suggeting.

Reply:
I meant it has no effect on visibility of the class itself. All these classes ending with underscore are - as far as I understand - internal implementation classes, but they are accessible from outside the package and the completion server suggests them during auto completion requests. This should not be the case. Since Haxe has no package visible modules, these classes should be implemented as private classes inside the public module, i.e. the module holding the public abstract.

Reply:
Well,for me a separate BigInt_ class leads to more readability, so I've added @:noCompletion and @:noDoc for now. If this is against the Haxe standart, I can put a BigInt_.hx class inside BigInt.hx ( and MutableBigInt_)